test/e2e/status_test.go

```diff
@@ -0,0 +1,148 @@
+	t.Run("VM configuration failure", testVMConfigurationFail)
+}
+
+// testVMProvisionFail tests that the status has the correct degradation reason on failure
+func testVMProvisionFail(t *testing.T) {
+	testCtx, err := NewTestContext(t)
+	require.NoError(t, err)
+	// create WMC custom resource with key pair that will fail
+	wmc, err := testCtx.createWMC(gc.numberOfNodes, "fakeKeyPair")
+	require.NoError(t, err, "error creating wcmo custom resource")
+	defer framework.Global.Client.Delete(context.TODO(), wmc)
+
+	degradedCondition, err := testCtx.waitForCondition(operator.Degraded, corev1.ConditionTrue)
+	require.NoError(t, err, "error waiting for degraded condition")
+
+	assert.Contains(t, degradedCondition.Reason, operator.VMCreationFailureReason)
+}
+
+// newCloudInterface returns an interface to interact with cloud resources
+func newCloudInterface() (cloudprovider.Cloud, error) {
+	kc := os.Getenv("KUBECONFIG")
+	if kc == "" {
+		return nil, fmt.Errorf("KUBECONFIG env var not set")
+	}
+	creds := os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
+	if creds == "" {
+		return nil, fmt.Errorf("AWS_SHARED_CREDENTIALS_FILE env var not set")
+	}
+	key := os.Getenv("KUBE_SSH_KEY_PATH")
+	if key == "" {
+		return nil, fmt.Errorf("KUBE_SSH_KEY_PATH")
+	}
+	tempDirPath, err := ioutil.TempDir(os.TempDir(), "wmco_test")
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create temp directory")
+	}
+
+	return cloudprovider.CloudProviderFactory(kc, creds, credentialAccountID, tempDirPath, "", "",
+		gc.sshKeyPair, key)
+}
+
+// testVMConfigurationFail tests that the status has the correct degradation reason on failure
+func testVMConfigurationFail(t *testing.T) {
+	testCtx, err := NewTestContext(t)
+	require.NoError(t, err)
+	// create WMC custom resource with key pair that will fail
+	wmc, err := testCtx.createWMC(gc.numberOfNodes, gc.sshKeyPair)
+	require.NoError(t, err, "error creating wcmo custom resource")
+	defer framework.Global.Client.Delete(context.TODO(), wmc)
+
+	err = testCtx.waitForWindowsNode(false)
+	require.NoError(t, err, "error waiting for windows node")
+	cloud, err := newCloudInterface()
+	require.NoError(t, err, "could not create cloud interface")
+
+	for _, node := range gc.nodes {
+		providerId := node.Spec.ProviderID
+		require.NotNil(t, providerId, "instance id unexpectedly nil")
+		splitID := strings.Split(providerId, "/")
+		require.Greaterf(t, len(splitID), 1, "provider ID %s was of unexpected format", providerId)
+		idToDelete := splitID[len(splitID)-1]
+		log.Printf("Deleting %s", idToDelete)
+		err = cloud.DestroyWindowsVM(idToDelete)
+		require.NoError(t, err, "error destroying Windows VM")
+		log.Printf("Deleted %s", idToDelete)
+	}
+
+	degradedCondition, err := testCtx.waitForCondition(operator.Degraded, corev1.ConditionTrue)
+	require.NoError(t, err, "error waiting for degraded condition")
+
+	assert.Contains(t, degradedCondition.Reason, operator.VMConfigurationFailureReason)
```

Comment:
Why is a VM being deleted out of band showing up as a configuration failure? I thought `VMConfigurationFailureReason` implied kubelet or CNI configuration issues. The VM itself being deleted wouldn't fall under that category.

Reply:
The VM is being deleted during the configuration causing the `nodeconfig.Configure()` function to fail. Any failure of that function counts as `VMConfigurationFailureReason`

Reply:
But how do you guarantee that it is getting deleted during `nodeconfig.Configure()`?

Reply:
Ya, it is not guaranteed :(. You perhaps need to wait for the node object to be created and see if the network is not yet and then induce the problem(the problem can be logging into the VM and stop hybrid overlay or delete the annotation). To  me, this seems an edge case to test which we don't want to produce in an e2e but a good unit test case in nodeConfig.go